### PR TITLE
unreal_pak upgrades

### PR DESCRIPTION
**Pull request by konsti219**

### CODE REVIEW

File: unreal_pak/src/buf_ext.rs [hunk 1]
```diff
@@ -1,11 +1,11 @@
 use std::io::{Read, Write};
 
-use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use byteorder::{ReadBytesExt, WriteBytesExt, LE};
 
 use crate::error::PakError;
 
 pub trait BufReaderExt {
-    fn read_fstring(&mut self) -> Result<Option<String>, PakError>;
+    fn read_fstring(&mut self) -> Result<String, PakError>;
```
Comment (localcc) [inline unreal_pak/src/buf_ext.rs:8]: maybe unify this function between uasset/unrealpak?

Comment (konsti219) [reply unreal_pak/src/buf_ext.rs:8]: going to do that after the PR is merged

File: unreal_pak/README.md [hunk 2]
```diff
@@ -1,3 +1,11 @@
 # unreal_pak
 
-Library crate for working with Unreal Engine .pak files. Currently only suuport version 8.
+Library crate for working with Unreal Engine .pak files.
+
+Features:
+
+- Supports reading and writing .pak files.
+- Can partially read files or load them entirely into memory depending on you API needs.
+- Aims to support all .pak versions, but currently only up to version 8.
```
Comment (localcc) [inline unreal_pak/README.md]: doesn't this PR add new versions?

File: unreal_pak/src/entry.rs [hunk 3]
```diff
@@ -27,13 +27,30 @@ where
 
     let header = Header::read(reader, pak_version)?;
 
-    match header.compression_method {
-        CompressionMethod::None => {
+    let compression_method = if pak_version >= PakVersion::FnameBasedCompressionMethod {
+        if header.compression_method == 0 {
```
[the remark this answers is not in the document]
Comment (konsti219) [reply unreal_pak/src/entry.rs:31]: I really don't want to have to add another Enum for compression...

File: unreal_pak/README.md (hunk 2 above)
Comment (konsti219) [reply unreal_pak/README.md]: fixed in new commit

File: unreal_pak/src/entry.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (localcc) [reply unreal_pak/src/entry.rs:31]: It would make more sense from code readability standpoint and also allowing only valid application states

**Review by localcc — APPROVED** [review]

**Pull request merged**